src/cascadia/TerminalSettingsEditor/LaunchViewModel.cpp

```diff
@@ -365,4 +368,88 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
     {
         return _Settings.DefaultTerminals();
     }
+
+    safe_void_coroutine LaunchViewModel::PrepareStartOnUserLoginSettings()
+    {
+        if (!StartOnUserLoginAvailable())
+        {
+            co_return;
+        }
+
+        auto strongThis{ get_strong() };
+        co_await winrt::resume_background();

```

Comment:
Any reason to not call this on the main thread?

Reply:
fixed!